src/content/ru/fundamentals/web-components/customelements.md

```diff
@@ -0,0 +1,1127 @@
+значений атрибутов, перечисленных в массиве `observedAttributes`.
+
+```
+class AppDrawer extends HTMLElement {
+  ...
+
+  static get observedAttributes() {
+    return ['disabled', 'open'];
+  }
+
+  get disabled() {
+    return this.hasAttribute('disabled');
+  }
+
+  set disabled(val) {
+    if (val) {
+      this.setAttribute('disabled', '');
+    } else {
+      this.removeAttribute('disabled');
+    }
+  }
+
+  // Only called for the disabled and open attributes due to observedAttributes
+  attributeChangedCallback(name, oldValue, newValue) {
+    // When the drawer is disabled, update keyboard/screen reader behavior.
+    if (this.disabled) {
+      this.setAttribute('tabindex', '-1');
+      this.setAttribute('aria-disabled', 'true');
+    } else {
+      this.setAttribute('tabindex', '0');
+      this.setAttribute('aria-disabled', 'false');
+    }
+    // TODO: also react to the open attribute changing.
+  }
+}
+```
+
+В этом примере мы задаем значения дополнительных атрибутов для `<app-drawer>`
+при изменении значения атрибута `disabled`. Хотя мы этого здесь не делаем, вы
+могли бы также **использовать `attributeChangedCallback` для синхронизации
+свойства элемента в JS с его атрибутом**.
+
+## Обновление элемента {: #upgrades}
+
+### Прогрессивно улучшенный HTML
+
+Мы уже узнали, что пользовательские элементы определяются при помощи вызова
+`customElements.define()`. Однако это не означает, что вы должны определить +
+зарегистрировать пользовательский элемент сразу.
+
+**Пользовательские элементы могут быть использованы *до* регистрации их
+определения** .
+
+Прогрессивное улучшение – возможность пользовательских элементов. Другими
+словами, вы можете объявить ряд элементов `<app-drawer>` на странице и вызвать
+`customElements.define('app-drawer', ...)` намного позже. Это так, поскольку
+браузер обрабатывает потенциальные пользовательские элементы иначе благодаря
+из-за возможности добавления тэгов с [неизвестными именами](#unknown). Процесс
+вызова `define()` и наделения существующего элемента определением класса
+называется «обновления элемента».
+
+Чтобы узнать, когда имя тэга становится определённым, вы можете использовать
+`window.customElements.whenDefined()`. Метод вернёт Promise, который выполнится
+когда элемент определится.
+
+```
+customElements.whenDefined('app-drawer').then(() => {
+  console.log('app-drawer defined');
+});
+```
+
+**Пример**: отложение выполнение кода до обновления дочерних элементов
+
+```
+<share-buttons>
+  <social-button type="twitter"><a href="...">Twitter</a></social-button>
+  <social-button type="fb"><a href="...">Facebook</a></social-button>
+  <social-button type="plus"><a href="...">G+</a></social-button>
+</share-buttons>
+
+
+
+// Fetch all the children of <share-buttons> that are not defined yet.
+let undefinedButtons = buttons.querySelectorAll(':not(:defined)');
+
+let promises = [...undefinedButtons].map(socialButton => {
+  return customElements.whenDefined(socialButton.localName);
+));
```

Comment:
Related to https://github.com/google/WebFundamentals/pull/8938/, should be:

`})`